src/exporter/ddprof_exporter.cc

```diff
@@ -140,49 +143,66 @@ DDRes ddprof_exporter_init(const ExporterInput *exporter_input,
   return ddres_init();
 }
 
-static void fill_tags(const UserTags *user_tags, const DDProfExporter *exporter,
-                      std::vector<ddprof_ffi_Tag> &tags_exporter) {
-  tags_exporter.push_back(ddprof_ffi_Tag{
-      .name = char_star_to_CharSlice("language"),
-      .value = string_view_to_CharSlice(exporter->_input.language)});
-
-  if (exporter->_input.environment) {
-    tags_exporter.push_back(ddprof_ffi_Tag{
-        .name = char_star_to_CharSlice("env"),
-        .value = char_star_to_CharSlice(exporter->_input.environment)});
+static DDRes add_single_tag_c(ddprof_ffi_Vec_tag &tags_exporter,
+                              const char *key, string_view value) {
+  ddprof_ffi_PushTagResult push_tag_res =
+      ddprof_ffi_Vec_tag_push(&tags_exporter, char_star_to_CharSlice(key),
+                              string_view_to_CharSlice(value));
+  defer { ddprof_ffi_PushTagResult_drop(push_tag_res); };
+  if (push_tag_res.tag == DDPROF_FFI_PUSH_TAG_RESULT_ERR) {
+    LG_ERR("[EXPORTER] Failure generate tag (%.*s)", (int)push_tag_res.err.len,
+           push_tag_res.err.ptr);
+    DDRES_RETURN_ERROR_LOG(DD_WHAT_EXPORTER, "Failed to generate tags");
   }
+  return ddres_init();
+}
 
-  if (exporter->_input.service_version) {
-    tags_exporter.push_back(ddprof_ffi_Tag{
-        .name = char_star_to_CharSlice("version"),
-        .value = char_star_to_CharSlice(exporter->_input.service_version)});
+static DDRes add_single_tag_cpp(ddprof_ffi_Vec_tag &tags_exporter,
+                                std::string_view key, std::string_view value) {
+  ddprof_ffi_PushTagResult push_tag_res =
+      ddprof_ffi_Vec_tag_push(&tags_exporter, cpp_string_to_CharSlice(key),
+                              cpp_string_to_CharSlice(value));
+  defer { ddprof_ffi_PushTagResult_drop(push_tag_res); };
+  if (push_tag_res.tag == DDPROF_FFI_PUSH_TAG_RESULT_ERR) {
+    LG_ERR("[EXPORTER] Failure generate tag (%.*s)", (int)push_tag_res.err.len,
+           push_tag_res.err.ptr);
+    DDRES_RETURN_ERROR_LOG(DD_WHAT_EXPORTER, "Failed to generate tags");
   }
+  return ddres_init();
+}
 
-  if (exporter->_input.service) {
-    tags_exporter.push_back(ddprof_ffi_Tag{
-        .name = char_star_to_CharSlice("service"),
-        .value = char_star_to_CharSlice(exporter->_input.service)});
-  }
+static DDRes fill_tags(const UserTags *user_tags,
+                       const DDProfExporter *exporter,
+                       ddprof_ffi_Vec_tag &tags_exporter) {
 
-  if (exporter->_input.profiler_version.len) {
-    tags_exporter.push_back(ddprof_ffi_Tag{
-        .name = char_star_to_CharSlice("profiler_version"),
-        .value = string_view_to_CharSlice(exporter->_input.profiler_version)});
-  }
+  DDRES_CHECK_FWD(
+      add_single_tag_cpp(tags_exporter, std::string_view("language"),
```

Comment:
No need to create `std::string_view`, it will be implicitly constructed from string literal.

Reply:
Indeed, Fixed.